app/code/core/Mage/Adminhtml/Controller/Action.php

```diff
@@ -395,13 +395,8 @@ protected function _validateSecretKey()
         if (is_array($this->_publicActions) && in_array($this->getRequest()->getActionName(), $this->_publicActions)) {
             return true;
         }
-
-        if (!($secretKey = $this->getRequest()->getParam(Mage_Adminhtml_Model_Url::SECRET_KEY_PARAM_NAME, null))
-            || !hash_equals(Mage::getSingleton('adminhtml/url')->getSecretKey(), $secretKey)
-        ) {
-            return false;
-        }
-        return true;
+        return !(!($secretKey = $this->getRequest()->getParam(Mage_Adminhtml_Model_Url::SECRET_KEY_PARAM_NAME, null))
+            || !hash_equals(Mage::getSingleton('adminhtml/url')->getSecretKey(), $secretKey));
```

Comment:
```suggestion
        return ($secretKey = $this->getRequest()->getParam(Mage_Adminhtml_Model_Url::SECRET_KEY_PARAM_NAME, null))
            && hash_equals(Mage::getSingleton('adminhtml/url')->getSecretKey(), $secretKey);
```

Reply:
Continue on the conversation on chaining rector rules, I don't like the idea that it requires another PR to clean up the code above. This example from rector is really bad. It should be clean up before committing. Don't you think?

Reply:
I'd prefer to revert last commit and add the DeMorgans rule separately.

Let's keep it simple. It only affects a few files.

Maybe add a new issue to not forget about it?

Reply:
I tried to apply the DeMorgan's rule to `!(!is_null($value) && !$value)`, not sure if I made the test correctly here: https://getrector.com/demo/6a07c84e-3a20-45c6-9513-d20329994eb8

Did I make a mistake in the test? It doesn't work, no code change.  If the above test is correct and if this PR is committed, there is no guarantee the convoluted code can be cleaned up by the rule later. So, now what?

Reply:
OK, so rector won't fix the convoluted bool expression now:
![image](https://github.com/user-attachments/assets/de678dbf-7596-412a-8dd5-41ee02f8023b)

So, should we apply DeMorgan's manually?

Reply:
https://github.com/OpenMage/magento-lts/pull/4152/commits/392d0a88c2a8c68854106c79288e727fa2c1660c

No. I'd take it as it is and apply that rule later. I dont think its worth to change it manually, same pattern is already there.